shared/lib/form.js

```diff
@@ -0,0 +1,90 @@
+/**
+ *! TODO: Extract into separate library module
```

Comment:
Is this already in library?

Reply:
No, I guess it'll stay like this until we have time for optimizations and details, so it's just a plan for this particular function I go once my form library is more mature.